ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -257,11 +258,17 @@ object WebSocketService {
             .toLeftDisjunction(x.ekey.templateId)
         }
 
-      val q: Map[domain.TemplateId.RequiredPkg, LfV] = resolvedWithKey.toMap
+      val q: Map[domain.TemplateId.RequiredPkg, HashSet[LfV]] =
+        resolvedWithKey.foldLeft(Map[domain.TemplateId.RequiredPkg, HashSet[LfV]]())((acc, el) =>
+          acc.get(el._1) match {
+            case Some(v) => acc.updated(el._1, v += el._2)
```

Comment:
As far as I can tell you are still appending instead of prepending which is quadratic. Can we prepend instead?

Reply:
This is a `HashSet` now so this is pretty much `O(1)`.

Reply:
Also note that this is a one-time construction at the point of establishing the stream. The one-time cost has changed from quadratic to linear, and the per-event cost (which matters more imo) is now "effectively constant" (i.e. logarithmic with a very large base), as per the documentation, rather than linear.

Reply:
Oh I missed that this is a HashSet now, makes sense then :+1: